src/main/java/at/hannibal2/skyhanni/features/misc/chocolatefactory/ChocolateFactoryStats.kt

```diff
@@ -0,0 +1,80 @@
+package at.hannibal2.skyhanni.features.misc.chocolatefactory
+
+import at.hannibal2.skyhanni.events.GuiRenderEvent
+import at.hannibal2.skyhanni.utils.NumberUtil.addSeparators
+import at.hannibal2.skyhanni.utils.RenderUtils.renderStrings
+import net.minecraftforge.fml.common.eventhandler.SubscribeEvent
+
+object ChocolateFactoryStats {
+
+    private val config get() = ChocolateFactoryApi.config
+
+    private var displayList = listOf<String>()
+
+    @SubscribeEvent
+    fun onBackgroundDraw(event: GuiRenderEvent.ChestGuiOverlayRenderEvent) {
+        if (!ChocolateFactoryApi.isEnabled()) return
+        if (!ChocolateFactoryApi.inChocolateFactory) return
+        if (!config.statsDisplay) return
+
+        config.position.renderStrings(displayList, posLabel = "Chocolate Factory Stats")
+    }
+
+    fun updateDisplay() {
+        val newList = mutableListOf<String>()
+        val perSecond = ChocolateFactoryApi.chocolatePerSecond
+        val perMinute = perSecond * 60
+        val perHour = perMinute * 60
+        val perDay = perHour * 24
+
+        newList.add("§6§lChocolate Factory Stats")
+
+        newList.add("§eCurrent Chocolate: §7${ChocolateFactoryApi.chocolateCurrent.addSeparators()}")
+        newList.add("§eThis Prestige: §7${ChocolateFactoryApi.chocolateThisPrestige.addSeparators()}")
+        newList.add("§eAll-time: §7${ChocolateFactoryApi.chocolateAllTime.addSeparators()}")
+
+        newList.add("§ePer Second: §7${perSecond.addSeparators()}")
+        newList.add("§ePer Minute: §7${perMinute.addSeparators()}")
+        newList.add("§ePer Hour: §7${perHour.addSeparators()}")
+        newList.add("§ePer Day: §7${perDay.addSeparators()}")
+
+        newList.add("§eChocolate Multiplier: §7${ChocolateFactoryApi.chocolateMultiplier}")
+        newList.add("§eBarn: §7${ChocolateFactoryBarnManager.barnStatus()}")
+
+        newList.add("")
+        newList.add("")
+        newList.add("")
```

Comment:
So, I don't like that yo take the list and then remap it. Better way is to add the static text to the enum and the dynamic one as () -> String, since the enum can call any of them. And the performance of doing the perDay, perHour, perMinute is nothing compared to remapping the list. Then the remapping is redundant since you can simply call displayList = config.statsDisplayList.map{it.print()} (print is only a example name here)

Reply:
I do agree with you but I think that can be a change for a later time as we do this in a few places rn and we can change all at the same time